IM_calculation/IM/im_calculation.py

```diff
@@ -445,23 +474,31 @@ def compute_measures_multiprocess(
                     sorted(components_to_calculate, key=lambda x: x.value),
                 )
             )
-        all_results.extend(p.starmap(compute_measure_single, array_params))
-
-    all_result_dict = ChainMap(*all_results)
-    output_path = get_result_filepath(output, identifier, ".csv")
-
-    results_dataframe = pd.DataFrame.from_dict(all_result_dict, orient="index")
-    results_dataframe.index = pd.MultiIndex.from_tuples(
-        results_dataframe.index, names=["station", "component"]
-    )
-    results_dataframe.sort_values(["station", "component"], inplace=True)
-    results_dataframe = order_im_cols_df(results_dataframe)
+            adv_array_params.append((waveform, advanced_im_config, output))
+        # only run basic im if and only if adv_im not going to run
+        if advanced_im_config.IM_list:
```

Comment:
We're checking IM_list before we get to checking if advanced_im_config is NULL. Can we move the check in compute_adv_measures_ here?

Reply:
What's the toughs behind this? i'm not against it, but doing so means that the function itself is depending on the which even parents script calls it.

Reply:
updated. logic moved one level above